elastic/kibana
# [Maps] Apply styles to icon SVGs in legend via descendent selectors

Fixes #128846

The wildcard descendent selector to applies fill and stroke to all elements in the SVG. This was not necessary before because Maki icons always had a single path element. But we don't have control over the number of elements with custom icons, so we must apply and override any fill and stroke settings on sibling and nested elements.

The `vector-effect: non-scaling-stroke` property fixes the issue where the stroke was barely visible in the legend with large custom icons.

Additionally, the !important property force overrides any style attributes in descendant elements. It is still possible that an element may have a style attribute with !important that we can not override. Hopefully that is a rare case though.

To test this:
 - Open the `[Flights] Origin Time Delayed` map from the Sample flight data
 - In Map settings, add a custom icon using an SVG with a width/height or viewBox much greater than 15px that also has style attributes applied to the descendent elements. This [Shoes walking](https://commons.wikimedia.org/wiki/Category:Black_SVG_icons#/media/File:Shoeprints_walking_icon.svg) is a good example.
  - Change the Flight Origin Location layer Symbol type to `icon` and select the custom icon
  - Change the Fill  and Border colors to solid colors. 
  - Verify the color in the Legend matches the selected colors.
